lib/matplotlib/text.py

```diff
@@ -276,86 +276,87 @@ def _get_layout(self, renderer):
         if key in self._cached:
             return self._cached[key]
 
-        horizLayout = []
-
         thisx, thisy = 0.0, 0.0
-        xmin, ymin = 0.0, 0.0
-        width, height = 0.0, 0.0
-        lines = self.get_text().split('\n')
+        lines = self.get_text().split('\n')  # Ensures lines is not empty.
 
-        whs = np.zeros((len(lines), 2))
-        horizLayout = np.zeros((len(lines), 4))
+        ws = []
+        hs = []
+        xs = []
+        ys = []
 
         # Full vertical extent of font, including ascenders and descenders:
-        tmp, lp_h, lp_bl = renderer.get_text_width_height_descent('lp',
-                                                         self._fontproperties,
-                                                         ismath=False)
-        offsety = (lp_h - lp_bl) * self._linespacing
+        _, lp_h, lp_d = renderer.get_text_width_height_descent(
+            'lp', self._fontproperties, ismath=False)
+        min_dy = (lp_h - lp_d) * self._linespacing
 
-        baseline = 0
         for i, line in enumerate(lines):
             clean_line, ismath = self.is_math_text(line, self.get_usetex())
             if clean_line:
-                w, h, d = renderer.get_text_width_height_descent(clean_line,
-                                                        self._fontproperties,
-                                                        ismath=ismath)
+                w, h, d = renderer.get_text_width_height_descent(
+                    clean_line, self._fontproperties, ismath=ismath)
             else:
-                w, h, d = 0, 0, 0
+                w = h = d = 0
 
-            # For multiline text, increase the line spacing when the
-            # text net-height(excluding baseline) is larger than that
-            # of a "l" (e.g., use of superscripts), which seems
-            # what TeX does.
+            # For multiline text, increase the line spacing when the text
+            # net-height (excluding baseline) is larger than that of a "l"
+            # (e.g., use of superscripts), which seems what TeX does.
             h = max(h, lp_h)
-            d = max(d, lp_bl)
+            d = max(d, lp_d)
 
-            whs[i] = w, h
+            ws.append(w)
+            hs.append(h)
 
+            # Metrics of the last line that are needed later:
             baseline = (h - d) - thisy
+
             if i == 0:
```

Comment:
```suggestion
            if thisy == 0:  # first line
```
gets rid of tne need for `enumerate()`.

Reply:
It's not immediately obvious that thisy cannot be zero for the second line too even if the first line is empty (it's true, but only thanks to the `h = max(h, lp_h)`/`d = max(d, lp_d)` lines) so I think leaving it as it is is clearer.

Reply:
Just by common sense, the vertical position `thisy` must increase also for empty lines. So it should only be zero before handling the first line.

But ok, if you want to leave it.

Reply:
I don't think common sense applies to this function (or to a lot of matplotlib, sadly).  (e.g. if you look at the way center_baseline is defined a bit below, it looks weirdly different between anchor and non-anchor rotations...)